fastavro/_schema.pyx

```diff
@@ -208,51 +230,32 @@ cdef _parse_schema(
 ):
     # union schemas
     if isinstance(schema, list):
-        parsed_schemas = []
-        for index, s in enumerate(schema):
-            if index == 0:
-                parsed_schemas.append(
-                    _parse_schema(
-                        s,
-                        namespace,
-                        expand,
-                        False,
-                        names,
-                        named_schemas,
-                        default,
-                        ignore_default_error,
-                        in_union=True,
```

Comment:
I think we can remove `in_union` from all function calls except for `_raise_default_value_error`, right? Previously it needed to be propagated everywhere because the default wasn't checked right away, but with these changes I don't think we need it in all the function signatures anymore.

Reply:
I went ahead and removed in_union entirely, since the only thing it was used for was switching the error message in _raise_default_value_error. That can also be done by checking whether the schema arg is a list.